lib/gyp.js

```diff
@@ -358,7 +358,7 @@ gyp.main = function main(args, extra) {
     gyp_binary: args[0],
     home_dot_gyp: homeDotGyp,
     root_targets: options.root_targets,
-    target_arch: cmdlineDefaultVariables['target_arch'] || ''
+    target_arch: cmdlineDefaultVariables['target_arch'] || process.arch
```

Comment:
As a reminder to myself, we should use `bindings.process` instead of global `process`.